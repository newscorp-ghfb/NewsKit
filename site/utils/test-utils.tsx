/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
