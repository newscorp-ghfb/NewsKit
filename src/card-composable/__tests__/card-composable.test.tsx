import React from 'react';
import {renderToFragmentWithTheme} from '../../test/test-utils';
import {
  CardComposable,
  CardMedia,
  CardContent,
  CardActions,
  CardLink,
} from '..';

describe('CardComposable', () => {
  test('renders defaults', () => {
    const fragment = renderToFragmentWithTheme(CardComposable, {
      children: (
        <>
          <CardContent>
            <CardLink href="/" expand>
              content
            </CardLink>
          </CardContent>
          <CardMedia media={{src: 'image/url.png', alt: 'image'}} />
          <CardActions>actions</CardActions>
        </>
      ),
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders without areas', () => {
    const fragment = renderToFragmentWithTheme(CardComposable, {
      areas: '',
      children: (
        <>
          <CardContent>
            <CardLink href="/">content</CardLink>
          </CardContent>
          <CardMedia media={{src: 'image/url.png', alt: 'image'}} />
          <CardActions>actions</CardActions>
        </>
      ),
    });
    expect(fragment).toMatchSnapshot();
  });
});
