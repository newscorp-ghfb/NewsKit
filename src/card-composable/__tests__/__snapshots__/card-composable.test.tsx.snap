// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComposable renders defaults 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 0;
  display: grid;
  grid-template-areas: "media" "content" "actions";
  color: #3B3B3B;
  background-color: #FFFFFF;
  position: relative;
}

.emotion-1 {
  margin: 0;
  padding: 0;
  display: grid;
  justify-items: start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-area: content;
}

.emotion-2 {
  display: inline-block;
  color: #3358CC;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: color,fill;
    transition-duration: 200ms,200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: color,fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2 svg {
  fill: #3358CC;
}

.emotion-2:hover:not(:disabled) {
  color: #254CAC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:hover:not(:disabled) svg {
  fill: #254CAC;
}

.emotion-2:active:not(:disabled) {
  color: #12387A;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:active:not(:disabled) svg {
  fill: #12387A;
}

.emotion-2:focus-visible:not(:disabled) {
  outline-color: #3768FB;
  outline-style: solid;
  outline-width: 2px;
  outline-offset: 2px;
}

@media not all and (min-resolution: 0.001dpcm) {
  @supports (-webkit-appearance: none) and (stroke-color: transparent) {
    .emotion-2:focus-visible:not(:disabled) {
      outline-style: auto;
    }
  }
}

.emotion-2:before {
  content: '';
  position: absolute;
  inset: 0;
  z-index: 1;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-4 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  display: block;
}

.emotion-4::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-4::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-5 {
  margin: 0;
  padding: 0;
  display: grid;
  grid-area: media;
}

.emotion-6 {
  position: relative;
  width: 100%;
  display: block;
  padding-top: 0;
  height: 0;
  width: 100%;
  border-radius: 0;
  background-color: #F1F1F1;
}

.emotion-6 svg {
  fill: #ABABAB;
}

.emotion-7 {
  top: 0;
  left: 0;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin: 0;
}

.emotion-8 {
  opacity: 0;
  display: block;
  border-radius: inherit;
  height: auto;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
}

.emotion-9 {
  margin: 0;
  padding: 0;
  display: grid;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  grid-area: actions;
  position: relative;
  z-index: 2;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        class="emotion-2"
        href="/"
      >
        <span
          class="emotion-3"
        >
          <span
            class="emotion-4"
          >
            content
          </span>
        </span>
      </a>
    </div>
    <div
      class="emotion-5"
    >
      <picture
        class="emotion-6"
      >
        <div
          class="emotion-7"
        />
        <img
          alt="image"
          class="emotion-8"
          src="image/url.png"
        />
      </picture>
    </div>
    <div
      class="emotion-9"
    >
      actions
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardComposable renders without areas 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  padding: 0;
  display: grid;
  color: #3B3B3B;
  background-color: #FFFFFF;
  position: relative;
}

.emotion-1 {
  margin: 0;
  padding: 0;
  display: grid;
  justify-items: start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-2 {
  display: inline-block;
  color: #3358CC;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media screen and (prefers-reduced-motion: no-preference) {
  .emotion-2 {
    transition-property: color,fill;
    transition-duration: 200ms,200ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

@media screen and (prefers-reduced-motion: reduce) {
  .emotion-2 {
    transition-property: color,fill;
    transition-duration: 0ms;
    transition-timing-function: cubic-bezier(0, 0, .5, 1),cubic-bezier(0, 0, .5, 1);
  }
}

.emotion-2 svg {
  fill: #3358CC;
}

.emotion-2:hover:not(:disabled) {
  color: #254CAC;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:hover:not(:disabled) svg {
  fill: #254CAC;
}

.emotion-2:active:not(:disabled) {
  color: #12387A;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2:active:not(:disabled) svg {
  fill: #12387A;
}

.emotion-2:focus-visible:not(:disabled) {
  outline-color: #3768FB;
  outline-style: solid;
  outline-width: 2px;
  outline-offset: 2px;
}

@media not all and (min-resolution: 0.001dpcm) {
  @supports (-webkit-appearance: none) and (stroke-color: transparent) {
    .emotion-2:focus-visible:not(:disabled) {
      outline-style: auto;
    }
  }
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

.emotion-4 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  display: block;
}

.emotion-4::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-4::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-5 {
  margin: 0;
  padding: 0;
  display: grid;
}

.emotion-6 {
  position: relative;
  width: 100%;
  display: block;
  padding-top: 0;
  height: 0;
  width: 100%;
  border-radius: 0;
  background-color: #F1F1F1;
}

.emotion-6 svg {
  fill: #ABABAB;
}

.emotion-7 {
  top: 0;
  left: 0;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  margin: 0;
}

.emotion-8 {
  opacity: 0;
  display: block;
  border-radius: inherit;
  height: auto;
  width: 100%;
  top: 0;
  left: 0;
  position: absolute;
}

.emotion-9 {
  margin: 0;
  padding: 0;
  display: grid;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: start;
  justify-content: start;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  position: relative;
  z-index: 2;
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        class="emotion-2"
        href="/"
      >
        <span
          class="emotion-3"
        >
          <span
            class="emotion-4"
          >
            content
          </span>
        </span>
      </a>
    </div>
    <div
      class="emotion-5"
    >
      <picture
        class="emotion-6"
      >
        <div
          class="emotion-7"
        />
        <img
          alt="image"
          class="emotion-8"
          src="image/url.png"
        />
      </picture>
    </div>
    <div
      class="emotion-9"
    >
      actions
    </div>
  </div>
</DocumentFragment>
`;
